docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,489 @@
+
+.. code-block:: bash
+
+    git clone https://github.com/Reference-LAPACK/lapack.git 
+    cd lapack; mkdir -p build; cd build 
+    cmake -DCMAKE_INSTALL_PREFIX=~/lapack -DCBLAS=ON -DLAPACK=ON -DLAPACKE=ON -DBUILD_INDEX64=ON -DBUILD_SHARED_LIBS=ON .. 
+    cmake --build . -j --target install 
+    cmake -DCMAKE_INSTALL_PREFIX=~/lapack -DCBLAS=ON -DLAPACK=ON -DLAPACKE=ON -DBUILD_INDEX64=OFF -DBUILD_SHARED_LIBS=ON .. 
+    cmake --build . -j --target install
+
+and then used in oneMKL by setting `-DREF_BLAS_ROOT=/path/to/lapack/install` and `-DREF_LAPACK_ROOT=/path/to/lapack/install`.
+
+When running tests you may encounter the issue ``BACKEND NOT FOUND EXECEPTION``, you may need to add your ``<oneMKL build directory>/lib`` to your ``LD_LIBRARY_PATH`` on Linux.
+
+.. _build_invocation_examples_dpcpp:
+
+CMake invocation examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Build oneMKL with support for x86 CPU, Intel GPU, and Nvidia GPUs with tests disabled using the Ninja build system:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \
+        -GNinja \
+        -DCMAKE_CXX_COMPILER=clang++ \
+        -DCMAKE_C_COMPILER=clang \
+        -DENABLE_MKLGPU_BACKEND=OFF \
+        -DENABLE_MKLCPU_BACKEND=OFF \
+        -DENABLE_CUFFT_BACKEND=ON \
+        -DENABLE_CUBLAS_BACKEND=ON \
+        -DENABLE_CUSOLVER_BACKEND=ON \
+        -DENABLE_CURAND_BACKEND=ON \
+        -DBUILD_FUNCTIONAL_TESTS=OFF
+
+`$ONEMKL_DIR` points at the oneMKL source directly.
+The x86 CPU (``MKLCPU``) and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled here.
+The backends for Nvidia GPUs must all be explicilty enabled.
+The tests are disabled, but the examples will still be built.
+
+Building oneMKL with support x86 CPU, Intel GPU, and AMD GPUs with tests disabled:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \
+        -DCMAKE_CXX_COMPILER=clang++ \
+        -DCMAKE_C_COMPILER=clang \
+        -DENABLE_MKLCPU_BACKEND=OFF \
+        -DENABLE_MKLGPU_BACKEND=OFF \
+        -DENABLE_ROCFFT_BACKEND=ON  \
+        -DENABLE_ROCBLAS_BACKEND=ON \
+        -DENABLE_ROCSOLVER_BACKEND=ON \
+        -DHIP_TARGETS=gfx90a \
+        -DBUILD_FUNCTIONAL_TESTS=OFF
+
+`$ONEMKL_DIR` points at the oneMKL source directly.
+The x86 CPU (``MKLCPU``) and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled here.
+The backends for AMD GPUs must all be explicilty enabled.
+The tests are disabled, but the examples will still be built.
+
+
+Build oneMKL for the DFT domain only with support for x86 CPU, Intel GPU, AMD GPU and Nvidia GPU with
+testing enabled:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \ 
+        -DCMAKE_CXX_COMPILER=icpx \
+        -DCMAKE_C_COMPILER=icx \ 
+        -DENABLE_ROCFFT_BACKEND=ON \
+        -DENABLE_CUFFT_BACKEND=ON \
+        -DTARGET_DOMAINS=dft \
+        -DBUILD_EXAMPLES=OFF
+
+Note that this is not a supported configuration, and requires Codeplay's oneAPI for AMD and Nvidia GPU plugins.
```

Comment:
Can you add links to the pages for the 2 plugins?

Reply:
Added in [90ce137](https://github.com/oneapi-src/oneMKL/pull/510/commits/90ce13728ac9a2a6a4ed4f969ca1ccb9de50e34b).